### PR TITLE
feat(adaptive_concurrency): support configuring the initial ARC limit

### PR DESCRIPTION
**Pull request by blake-mealey**
Following [discussion in the discord](https://discord.com/channels/742820443487993987/1134176002663252108), adds a new `initial_concurrency` option to the ARC settings, and defaults the ARC limit to that value when available.

### CODE REVIEW

File: src/sinks/util/adaptive_concurrency/mod.rs [hunk 1]
```diff
@@ -29,6 +29,14 @@ pub(self) fn instant_now() -> std::time::Instant {
 #[derive(Clone, Copy, Debug)]
 #[serde(deny_unknown_fields)]
 pub struct AdaptiveConcurrencySettings {
+    /// The initial concurrency limit to use. If not specified, the initial limit will be 1 (no concurrency).
```
Comment (dsmith3197) [inline src/sinks/util/adaptive_concurrency/mod.rs:32]: Can you execute `make generate-component-docs` to update the docs as well?

**Review by dsmith3197 — COMMENTED** [review]
Looks good! Will approve once docs are updated.

File: src/sinks/util/adaptive_concurrency/mod.rs (hunk 1 above)
Comment (blake-mealey) [reply src/sinks/util/adaptive_concurrency/mod.rs:32]: Updated!

**Review by dsmith3197 — CHANGES_REQUESTED** [review]

File: src/sinks/util/adaptive_concurrency/mod.rs [hunk 2]
```diff
@@ -29,6 +29,14 @@ pub(self) fn instant_now() -> std::time::Instant {
...
+    ///
+    /// It is recommended to set this value to your service's average limit if you're seeing that it takes a
+    /// long time to ramp up adaptive concurrency after a restart. You can find this value by looking at the
+    /// `adaptive_concurrency_limit` metric.
+    #[configurable(derived)]
+    pub(super) initial_concurrency: Option<usize>,
+
```
[the remark this answers is not in the document]
Comment (blake-mealey) [reply src/sinks/util/adaptive_concurrency/mod.rs]: I was unable to pass `MAX_CONCURRENCY` (or `"MAX_CONCURRENCY"`) to the validation range, so I instead passed `200` directly and added a comment to the `MAX_CONCURRENCY` constant that any change to the constant should also be made to this range.

**Review by dsmith3197 — APPROVED** [review]
Looks great, thanks for the contribution @blake-mealey!

**Comment by blake-mealey** [comment]
Thanks for approving! When do you think this change will be released?

**Comment by jszwedko** [comment]
> Thanks for approving! When do you think this change will be released?

It should be in 0.32.0 next week.

**Comment by blake-mealey** [comment]
Amazing, thank you!

**Pull request merged**